Source/WebKit/NetworkProcess/storage/BackgroundFetchStoreImpl.cpp

```diff
@@ -294,6 +308,72 @@ void BackgroundFetchStoreImpl::retrieveResponseBody(const ServiceWorkerRegistrat
     });
 }
 
+void BackgroundFetchStoreImpl::fetchInformationFromFilename(const String& filename, CompletionHandler<void(const ServiceWorkerRegistrationKey&, const String&)>&& callback)
+{
+    loadAllFetches([protectedThis = Ref { *this }, filename, callback = WTFMove(callback)]() mutable {
+        auto iterator = protectedThis->m_filenameToFetch.find(filename);
+        if (iterator == protectedThis->m_filenameToFetch.end()) {
+            callback({ }, { });
+            return;
+        }
+
+        ServiceWorkerRegistrationKey key { SecurityOriginData { iterator->value.origin.topOrigin }, URL { { }, iterator->value.identifier.first } };
+        callback(key, iterator->value.identifier.second);
+    });
+}
+
+void BackgroundFetchStoreImpl::loadAllFetches(CompletionHandler<void()>&& callback)
+{
+    m_manager->getAllOrigins([protectedThis = Ref { *this }, callback = WTFMove(callback)](auto&& origins) mutable {
```

Comment:
Right, I'll change to get this info from SWServer, it will only load origins that have service workers.
We should also probably update dispatchTaskToBackgroundFetchManager to add an option whether to get a bg fetch manager if there is no corresponding directory